dev/devicelab/test/adb_test.dart

```diff
@@ -166,15 +166,14 @@ class CommandArgs {
   }
 
   @override
-  int get hashCode => 17 * (17 * command.hashCode + _hashArguments) + _hashEnvironment;
-
-  int get _hashArguments => arguments != null
-    ? const ListEquality<String>().hash(arguments)
-    : null.hashCode;
-
-  int get _hashEnvironment => environment != null
-    ? const MapEquality<String, String>().hash(environment)
-    : null.hashCode;
+  int get hashCode {
+    return Object.hash(
+      command,
+      Object.hashAll(arguments ?? const <String>[]),
+      Object.hashAllUnordered(environment?.keys ?? const <String>[]),
```

Comment:
Not that it matters too much, but what does `hashAllUnordered` do here over hash all?

Reply:
Order independent hashing. https://github.com/dart-lang/sdk/commit/83376bf1ee81607a182f8558b242f1c9f4883246#diff-dc1389d26b294bb9804491e03fdf84c939c81eddc77578a0d90417b052f2a3c8R524

It's probably not super necessary here, where the map is always going to be a LinkedHashMap and preserve order, but I did it for the other map so I figured why not here too.